tc-health-client/tmagent/tmagent.go

```diff
@@ -656,7 +656,9 @@ func (pi *ParentInfo) GetTOData(cfg *config.Cfg) error {
 		if *sv.CDNName != cfg.CDNName {
 			continue
 		}
-		if sv.Type == tc.MonitorTypeName && tc.CacheStatus(*sv.Status) == tc.CacheStatusOnline {
+		// added OR because MonitorTypeName has been updated to 'TRAFFIC_MONITOR' and TO still has then listed as 'RASCAL'
+		// this will allow it to work either way and can be removed once TO is updated.
+		if (sv.Type == tc.MonitorTypeName || sv.Type == "RASCAL") && tc.CacheStatus(*sv.Status) == tc.CacheStatusOnline {
```

Comment:
The `sv.Type` comparison should be against a shared constant like `tc.MonitorTypeName` rather than the string literal `"RASCAL"`.

https://github.com/apache/trafficcontrol/blob/1c26082ebff759f4311845caee0d5cc1e0a4baa4/lib/go-tc/enum.go#L134-L136

- It's unfortunate that `tc.MonitorTypeName`'s documentation appears to discourages its use. It was added in #6040, and if I had noticed it at the time, I would have suggested rewording it before merging #6040. I think what it's trying to say is that its value could potentially be updated to something else in a future PR, but @ocket8888 can tell us for sure what he was trying to say.

- `tc.MonitorTypeName` changed from `"RASCAL"` to `"TRAFFIC_MONITIOR"` in #7068

Reply:
Only compared against shared constant.

Reply:
I just read the comment (and noticed the rest of the code line, which includes `tc.MonitorTypeName` anyway).

[this will allow it to work either way and can be removed once TO is updated.](https://github.com/apache/trafficcontrol/blob/1c26082ebff759f4311845caee0d5cc1e0a4baa4/tc-health-client/tmagent/tmagent.go#L660)

I agree that checking for both `"TRAFFIC_MONITIOR"` and `"RASCAL"` is useful here, but I still think only shared constants should be used. How about making a new constant for "`RASCAL"`? Not sure how to version it, but there's def value in keeping the old one around to help with upgrades